mobility_games/games/ar_find.py

```diff
@@ -0,0 +1,217 @@
+                        self.tag_array)
+        self.x = None   #   x and y position of Tango
+        self.y = None
+        self.z = None
+        self.tag_list = []
+        self.first_tag = None
+        self.tag_x = None
+        self.tag_y = None
+        self.tag_z = None
+        self.yaw = None
+        self.start = False
+        self.listener = tf.TransformListener()  #   Starts listening
+        self.soundtrack = song_sample_1()
+        self.quarter_time = 0.25
+        self.player = player()
+        self.tag_dict = {}
+
+    def process_pose(self, msg):    #   Updates position of Tango
+        self.x = msg.pose.position.x
+        self.y = msg.pose.position.y
+        self.z = msg.pose.position.z
+        angles = euler_from_quaternion([msg.pose.orientation.x,
+                                        msg.pose.orientation.y,
+                                        msg.pose.orientation.z,
+                                        msg.pose.orientation.w])
+        self.yaw = angles[2]
+
+    def tag_array(self, msg):
+        # Processes the april tags currently in Tango's view.
+        # Adds a four-integer tuple to self.tag_list for each tag detected,
+        # which includes x, y, and z coordinates and the tag id.
+        self.tag_list = []
+        self.orientation_list = []
+        self.id_list = []
+        try:
+            for item in msg.detections:
+                newitem = self.listener.transformPose('odom', item.pose)
+                x = newitem.pose.position.x
+                y = newitem.pose.position.y
+                z = newitem.pose.position.z
+                tag_id = item.id
+                self.tag_list.append((x, y, z, tag_id))
+                self.tag_dict[tag_id] = (x, y, z)
+                angles = euler_from_quaternion([newitem.pose.orientation.x,
+                                                newitem.pose.orientation.y,
+                                                newitem.pose.orientation.z,
+                                                newitem.pose.orientation.w])
+                self.orientation_list.append(angles)
+                self.id_list.append(tag_id)
+        except:
+            for item in msg.detections:
+                x = None
+                y = None
+                z = None
+                tag_id = item.id
+                self.tag_list.append((x, y, z, tag_id))
+                self.id_list.append(tag_id)
+
+    def run(self):
+        r = rospy.Rate(10)
+        print("Searching for Tango...")
+        while not rospy.is_shutdown():
+            if self.x and not self.start:
+                self.start = True
+                print("Connection established.")
+                print("Searching for tags...")
+            a = []
+            list_of_visible_tags = [cell[3] for cell in self.tag_list]
+            if len(self.tag_list) and self.start:
+                if not self.first_tag and not None in [item[1] for item in self.tag_list]:
```

Comment:
The use of Y is arbitrary. Should I address this with a comment? Would it be more intuitive to use index 0?